sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -0,0 +1,433 @@
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        ///   Converts the instance to string representation.
+        /// </summary>
+        ///
+        /// <returns>A <see cref="System.String" /> that represents this instance.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString() => $"Event Processor<{ typeof(TPartition).Name }>: { Identifier }";
+
+        /// <summary>
+        ///   Produces a list of the available checkpoints for the Event Hub and consumer group associated with the
+        ///   event processor instance, so that processing for a given set of partitions can be properly initialized.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the processing.  This is most likely to occur when the processor is shutting down.</param>
+        ///
+        /// <returns>The set of checkpoints for the processor to take into account when initializing partitions.</returns>
+        ///
+        /// <remarks>
+        ///   Should a partition not have a corresponding checkpoint, the <see cref="EventProcessorOptions.DefaultStartingPosition" /> will
+        ///   be used to initialize the partition for processing.
+        ///
+        ///   In the event that a custom starting point is desired for a single partition, or each partition should start at a unique place,
+        ///   it is recommended that this method express that intent by returning checkpoints for those partitions with the desired custom
+        ///   starting location set.
+        /// </remarks>
+        ///
+        protected abstract Task<IEnumerable<EventProcessorCheckpoint>> ListCheckpointsAsync(CancellationToken cancellationToken);
+
+        /// <summary>
+        ///   Produces a list of the ownership assignments for partitions between each of the cooperating event processor
+        ///   instances for a given Event Hub and consumer group pairing.  This method is used when load balancing to allow
+        ///   the processor to discover other active collaborators and to made decisions about how to best balance work
+        ///   between them.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the processing.  This is most likely to occur when the processor is shutting down.</param>
+        ///
+        /// <returns>The set of ownership records to take into account when making load balancing decisions.</returns>
+        ///
+        protected abstract Task<IEnumerable<EventProcessorPartitionOwnership>> ListOwnershipAsync(CancellationToken cancellationToken);
+
+        /// <summary>
+        ///   Attempts to claim ownership of the specified partitions for processing.  This method is used when
```

Comment:
"This method is used *by*"

Reply:
Thanks!   Most of these were written in chunks between meetings.   I'm expecting a bunch of typos and "this doesn't sound quite right" to come up.   :smile: